paddle/fluid/operators/listen_and_serv_op.cc

```diff
@@ -238,8 +249,20 @@ void ListenAndServOp::RunAsyncLoop(framework::Executor *executor,
     grad_to_prepared_ctx[id_to_grad[block_list[i]]] = optimize_prepared[i];
   }
 
-  VLOG(3) << "RunAsyncLoop into while";
   bool exit_flag = false;
+
+  VLOG(3) << "start async optimize threads";
+  std::vector<std::future<void>> fs;
+  for (auto iter = grad_to_queue.begin(); iter != grad_to_queue.end(); iter++) {
```

Comment:
Need to use a thread pool here to control the size of threads running blocks, for the number of CPU cores may be much less than grad vars, this may reduce the performance.